patches/server/0979-Don-t-tab-complete-namespaced-commands-if-send-names.patch

```diff
@@ -0,0 +1,28 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: EpicPlayerA10 <adasko.lesiak@gmail.com>
+Date: Sun, 18 Jun 2023 12:38:24 +0200
+Subject: [PATCH] Don't tab-complete namespaced commands if send-namespaced is
+ false
+
+Tab-complete packet is supposed to tab-complete args for commands, but
+it also can suggest commands like in version 1.12.2 or lower.
+
+This patch prevents server from sending namespaced commands when player
+requests tab-complete only commands.
+
+diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+index 2a609e43370e68943c580083f7f7d8c9b0972955..6c1d164561b9b7b4d905832fb65e9b34d21b4de7 100644
+--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
++++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+@@ -918,6 +918,11 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
+                         ParseResults<CommandSourceStack> parseresults = this.server.getCommands().getDispatcher().parse(stringreader, this.player.createCommandSourceStack());
+ 
+                         this.server.getCommands().getDispatcher().getCompletionSuggestions(parseresults).thenAccept((suggestions) -> {
++                            // Paper start - Don't tab-complete namespaced commands if send-namespaced is false
++                            if (!org.spigotmc.SpigotConfig.sendNamespaced && !command.contains(" ")) {
```

Comment:
I'm not 100% sure on how the brig `Suggestions` type works, but could you instead check the string range in the `Suggestions` object to see if it begins at 0? That might be a better way of knowing if its completing the first node.

Reply:
I don't know if I understood it correctly. What do you mean by checking the string range? What string range?

Reply:
The Suggestions object in that completable future contains a range that the suggestions is for.

Reply:
Done